internal/providers/pluginfw/resources/cluster/tests/acceptance/data_cluster_test.go

```diff
@@ -0,0 +1,45 @@
+package acceptance
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/databricks/terraform-provider-databricks/internal/acceptance"
+)
+
+var sparkVersionDataSourceTemplate = `
+	data "databricks_spark_version" "latest" {}
+`
+
+func clusterResourceTemplate(resourceName string, clusterName string) string {
+	return fmt.Sprintf(`
+		resource "databricks_cluster" "%s" {
+			cluster_name = "%s"
+			instance_pool_id = "{env.TEST_INSTANCE_POOL_ID}"
+			spark_version = data.databricks_spark_version.latest.id
+			num_workers = 1
+			autotermination_minutes = 10
+		}
+	`, resourceName, clusterName)
+}
+
+func TestAccDataSourceClusterByID(t *testing.T) {
+	acceptance.WorkspaceLevel(t, acceptance.Step{
+		Template: `
+		data "databricks_cluster_pluginframework" "this" {
+			cluster_id = "{env.TEST_DEFAULT_CLUSTER_ID}"
+		}`,
+	})
+}
+
+func TestAccDataSourceClusterByName(t *testing.T) {
+	clusterName := acceptance.RandomName("cluster-test-name-")
+	acceptance.WorkspaceLevel(t, acceptance.Step{
+		Template: sparkVersionDataSourceTemplate + clusterResourceTemplate("this", clusterName),
+	}, acceptance.Step{
+		Template: sparkVersionDataSourceTemplate + clusterResourceTemplate("this", clusterName) + fmt.Sprintf(`
+		data "databricks_cluster_pluginframework" "this" {
+			cluster_name = "%s"
+		}`, clusterName),
+	})
```

Comment:
Suggestion: instead of creating a new cluster (which is kind of slow), why don't we just use the cluster name of the test_default_cluster_id that we just fetched?

Reply:
We don’t currently export them in env variables in our test infra so in cases the name changes (which is highly unlikely). I think going ahead with relying on default cluster’s name to be "DEFAULT Test Cluster"  should be good. It would be good to have an environment variable for this as well. I will create a tracking ticket for that.